[threaded-animation-resolution] refactor code indicating whether two filter lists can be interpolated to be accessible outside of CSSPropertyAnimation

graouts opened the pull request:
#### e8e6ceec1f12b02e96b0a21144449934234a0049
<pre>
[threaded-animation-resolution] refactor code indicating whether two filter lists can be interpolated to be accessible outside of CSSPropertyAnimation
<a href="https://bugs.webkit.org/show_bug.cgi?id=268534">https://bugs.webkit.org/show_bug.cgi?id=268534</a>

Reviewed by Simon Fraser.

As part of the threaded animation resolution work (bug 250970), we will need to be able to identify
cases where `filter` values cannot be interpolated because it will be both difficult to implement
animation of mis-matching filter lists using the Core Animation API and also because it doesn&apos;t
make much sense to run an animation in the UIProcess for values that will not continuously change.

The logic that determines whether two `filter` values can be interpolated is private to `CSSPropertyAnimation`
as it stands. This change exposes it on `FilterOperations` directly which will allow `AcceleratedEffectStack`
as well as its remote counterpart to reason about whether a `filter` animation ought to be accelerated.

* Source/WebCore/animation/CSSPropertyAnimation.cpp:
* Source/WebCore/platform/graphics/filters/FilterOperations.cpp:
(WebCore::FilterOperations::canInterpolate const):
* Source/WebCore/platform/graphics/filters/FilterOperations.h:

Canonical link: <a href="https://commits.webkit.org/273932@main">https://commits.webkit.org/273932@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ac4fc328f49dacf518598416d89e62a4caa43765

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/37275 "Passed style check") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/48/builds/16172 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/14/builds/39566 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/39832 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/33247 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [  ~~🧪 bindings~~](https://ews-build.webkit.org/#/builders/9/builds/38579 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/18758 "Built successfully") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/51/builds/13317 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/39832 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [  ~~🧪 webkitperl~~](https://ews-build.webkit.org/#/builders/11/builds/37840 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/13640 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/14/builds/39566 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/39832 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/11864 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/14/builds/39566 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/41092 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/33744 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/14/builds/39566 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/41092 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/12216 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/51/builds/13317 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/41092 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/13868 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/14/builds/39566 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/12567 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/4819 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/13120 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

smfr commented on Source/WebCore/platform/graphics/filters/FilterOperations.cpp:
```diff
@@ -175,6 +175,33 @@ bool FilterOperations::hasFilterThatShouldBeRestrictedBySecurityOrigin() const
     return false;
 }
 
+bool FilterOperations::canInterpolate(const FilterOperations& to, CompositeOperation compositeOperation) const
```
Does this apply to all implementations, or does it describe specific limitations of Core Animation based ones? If so, it should live in platform code.

graouts replied:
No, this is the rule as specified by CSS Filters, the link is just a little below.

graouts committed:
[threaded-animation-resolution] refactor code indicating whether two filter lists can be interpolated to be accessible outside of CSSPropertyAnimation
https://bugs.webkit.org/show_bug.cgi?id=268534

Reviewed by Simon Fraser.

As part of the threaded animation resolution work (bug 250970), we will need to be able to identify
cases where `filter` values cannot be interpolated because it will be both difficult to implement
animation of mis-matching filter lists using the Core Animation API and also because it doesn't
make much sense to run an animation in the UIProcess for values that will not continuously change.

The logic that determines whether two `filter` values can be interpolated is private to `CSSPropertyAnimation`
as it stands. This change exposes it on `FilterOperations` directly which will allow `AcceleratedEffectStack`
as well as its remote counterpart to reason about whether a `filter` animation ought to be accelerated.

* Source/WebCore/animation/CSSPropertyAnimation.cpp:
* Source/WebCore/platform/graphics/filters/FilterOperations.cpp:
(WebCore::FilterOperations::canInterpolate const):
* Source/WebCore/platform/graphics/filters/FilterOperations.h:

Canonical link: https://commits.webkit.org/273932@main
